File: C++/Array/2279_Maximum Bags With Full Capacity of Rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        
        int n = capacity.size();
        
        vector<int> diff(n,0);
        
        for(int i = 0;i<n;i++){
            diff[i] = capacity[i] - rocks[i];
        }
        
        int count = 0;
        sort(begin(diff), end(diff));
        
        for(int i = 0;i<n;i++){
            if(additionalRocks >= diff[i]){
                additionalRocks -= diff[i];
                count++;
            }
            else{
                break;
            }
        }
        
        return count;
    }
};

File: Python/Array/2279_Maximum Bags With Full Capacity of Rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        
        rem_cap = [cap-roc for cap, roc in zip(capacity, rocks)]
        
        rem_cap.sort();
        
        full = 0
        
        for curr in rem_cap:
            if additionalRocks >= curr:
                additionalRocks -= curr
                full +=1
            else:
                break
        
        
        return full